packages/mds-transactions-service/repository/mappers.ts

```diff
@@ -1,27 +1,99 @@
 import { Optional, Timestamp } from '@mds-core/mds-types'
 import { IdentityColumn, ModelMapper, RecordedColumn } from '@mds-core/mds-repository'
 import { TransactionEntityModel } from './entities/transaction-entity'
-import { TransactionDomainCreateModel, TransactionDomainModel } from '../@types'
+import { TransactionOperationEntityModel } from './entities/operation-entity'
+import { TransactionStatusEntityModel } from './entities/status-entity'
+import {
+  TransactionDomainCreateModel,
+  TransactionDomainModel,
+  TransactionOperationDomainCreateModel,
+  TransactionOperationDomainModel,
+  TransactionStatusDomainCreateModel,
+  TransactionStatusDomainModel
+} from '../@types'
 
 type TransactionEntityToDomainOptions = Partial<{}>
 
-export const TransactionEntityToDomain = ModelMapper<TransactionEntityModel, TransactionDomainModel, TransactionEntityToDomainOptions>(
-  (entity, options) => {
-    const { id, recorded, ...domain } = entity
-    return domain
-  }
-)
+export const TransactionEntityToDomain = ModelMapper<
+  TransactionEntityModel,
+  TransactionDomainModel,
+  TransactionEntityToDomainOptions
+>((entity, options) => {
+  const { id, recorded, ...domain } = entity
+  return domain
+})
+
+type TransactionOperationEntityToDomainOptions = Partial<{}>
+
+export const TransactionOperationEntityToDomain = ModelMapper<
+  TransactionOperationEntityModel,
+  TransactionOperationDomainModel,
+  TransactionOperationEntityToDomainOptions
+>((entity, options) => {
+  const { id, recorded, ...domain } = entity
+  return domain
+})
+
+type TransactionStatusEntityToDomainOptions = Partial<{}>
+
+export const TransactionStatusEntityToDomain = ModelMapper<
+  TransactionStatusEntityModel,
+  TransactionStatusDomainModel,
+  TransactionStatusEntityToDomainOptions
+>((entity, options) => {
+  const { id, recorded, ...domain } = entity
+  return domain
+})
 
 type TransactionEntityCreateOptions = Partial<{
   recorded: Timestamp
 }>
 
-export type TransactionEntityCreateModel = Omit<Optional<TransactionEntityModel, keyof RecordedColumn>, keyof IdentityColumn>
+export type TransactionEntityCreateModel = Omit<
+  Optional<TransactionEntityModel, keyof RecordedColumn>,
+  keyof IdentityColumn
+>
 
 export const TransactionDomainToEntityCreate = ModelMapper<
   TransactionDomainCreateModel,
   TransactionEntityCreateModel,
   TransactionEntityCreateOptions
+>(({ device_id = null, ...domain }, options) => {
+  const { recorded } = options ?? {}
```

Comment:
If `options` is `null` or `undefined`, then you'll be trying to destructure `recorded` from an empty object. I just tried that in the node console and it doesn't seem happy:

```
> const { f } = {}
[eval].ts:18:9 - error TS2525: Initializer provides no value for this binding element and the binding element has no default value.
```

Reply:
By "node console" do you mean `ts-node`? That's a typing error, not a runtime error. The TS compiler works just fine and actually generates something like `const { recorded } = options !== null && options !== void 0 ? options : {}` and it knows that `options` (and `{}` in this context) is a `Partial` so it doesn't require `recorded` to exist on the object. There's nothing wrong with this code.